mqtt_io/modules/stream/serial.py

```diff
@@ -80,9 +95,22 @@ def setup_module(self) -> None:
         self.ser.flushInput()
 
     def read(self) -> Optional[bytes]:
-        return self.ser.read(self.ser.in_waiting) or None
+        data = None
+        if "delimiter" in self.config:
+            if "reset_before_read" in self.config:
+                if self.config["reset_before_read"]:
+                    self.ser.reset_input_buffer()
+            data = self.ser.read_until(self.config["delimiter"].encode("utf-8"))
+            if data:
+                data = data[:-len(self.config["delimiter"])]
+        else:
+            data = self.ser.read(self.ser.in_waiting) or None
```

Comment:
@jtkDvlp I guess this is the problem:
read has to return Optinal[bytes], but in line 107 it may return "None" (or with the preset in line 98)